lib/inspec/plugin/v2/installer.rb

```diff
@@ -368,9 +377,19 @@ def install_gem_to_plugins_dir(new_plugin_dependency, # rubocop: disable Metrics
       # of the same gem on the stack or a malformed dependency.
       begin
         solution.each do |activation_request|
-          unless activation_request.full_spec.activated?
-            activation_request.full_spec.activate
+          requested_gemspec = activation_request.full_spec
```

Comment:
Can you add some comments over this block around it's functionality? Or the use cases this is handling? I know this might be old code.. but since we have got into it now..Will be good for future ref to understand code later.

Reply:
I mean if somebody has to understand this code they have to dig into this either way .. but to make this process easier I shall add some comments

Reply:
Thanks!